compiler/src/dotty/tools/dotc/typer/Applications.scala

```diff
@@ -333,7 +333,7 @@ object Applications {
       // it's crucial that the type tree is not copied directly as argument to
       // `cpy$default$1`. If it was, the variable `X'` would already be interpolated
       // when typing the default argument, which is too early.
-      spliceMeth(meth, fn).appliedToTypes(targs.tpes)
+      spliceMeth(meth, fn).appliedToTypeTrees(targs.map(targ => TypeTree(targ.tpe).withSpan(targ.span)))
```

Comment:
Could this type of confusion occur somewhere else in code, but we just didn't notice it yet?

Reply:
I don't understand the question.  By code, do you mean compiler code, or do you mean something about user's code?

Reply:
Not that I'm aware of.  Looking now, perhaps the `.appliedToTypes(patternTypes)` in quotes pattern should reuse the original pattern tree spans, but I'm not certain.  I was thinking Martin would be able to tell if this is a reoccurring pattern and how to best deal with it.